Gems/Atom/Tools/MaterialCanvas/Code/Source/Window/MaterialCanvasGraphView.cpp

```diff
@@ -93,13 +89,13 @@ namespace MaterialCanvas
         nodePaletteConfig.m_mimeType = "materialcanvas/node-palette-mime-event";
         nodePaletteConfig.m_isInContextMenu = true;
         nodePaletteConfig.m_saveIdentifier = "MaterialCanvas_ContextMenu";
-        nodePaletteConfig.m_rootTreeItem = GetNodePaletteRootTreeItem();
+        nodePaletteConfig.m_rootTreeItem = m_createNodePaletteItemsCallback(m_toolId);
         m_sceneContextMenu = aznew GraphCanvas::SceneContextMenu(m_toolId, this);
         m_sceneContextMenu->AddNodePaletteMenuAction(nodePaletteConfig);
 
         // Setup the context menu with node palette for proposing a new node
         // when dropping a connection in an empty space in the graph
-        nodePaletteConfig.m_rootTreeItem = GetNodePaletteRootTreeItem();
+        nodePaletteConfig.m_rootTreeItem = m_createNodePaletteItemsCallback(m_toolId);
```

Comment:
Why is this line repeated? The same thing was done at line 92

Reply:
Two different context menus containing embedded trees are being populated and presumably take ownership of the pointers that get returned from the function. This function is called a third time for the docked node palette window. A different or parameterized function could be used to generate a different hierarchy.